### ISSUE TITLE
"What libraries should I install to use PyTorch and YOLOv5? I am using PyCharm and Visual Studio."

### ISSUE THREAD

**Issue by joseant7231** [ultralytics/yolov5#13579]
### Search before asking

- [x] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Good morning, good afternoon, or good evening. I am using Visual Studio Code and also PyCharm with the Python language. I would like to perform image recognition on a video, but I am encountering errors with Torch and YOLOv5. Could you please help me by telling me which libraries I should install first? Thank you very much for your possible response.

### Additional

![Image](https://github.com/user-attachments/assets/6db9e84d-0ae2-421f-83f5-08e92d55f5d6)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @joseant7231, thank you for your interest in YOLOv5 🚀! This is an automated response to help you get started quickly—an Ultralytics engineer will also assist you soon.

Please visit our ⭐️ [Tutorials](https://docs.ultralytics.com/yolov5/) for detailed guides—including [Custom Data Training](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data/) and [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution/).

If this is a 🐛 Bug Report, please provide a **minimum reproducible example** to help us debug it.

If this is a custom training ❓ Question, please include as much information as possible, such as dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

## Requirements

[**Python>=3.8.0**](https://www.python.org/) with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) installed including [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). To get started:
```bash
git clone https://github.com/ultralytics/yolov5  # clone
cd yolov5
pip install -r requirements.txt  # install
```

## Environments

YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://bit.ly/yolov5-paperspace-notebook"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml"><img src="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml/badge.svg" alt="YOLOv5 CI"></a>

If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are currently passing. CI tests verify correct operation of YOLOv5 [training](https://github.com/ultralytics/yolov5/blob/master/train.py), [validation](https://github.com/ultralytics/yolov5/blob/master/val.py), [inference](https://github.com/ultralytics/yolov5/blob/master/detect.py), [export](https://github.com/ultralytics/yolov5/blob/master/export.py) and [benchmarks](https://github.com/ultralytics/yolov5/blob/master/benchmarks.py) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

😊 If you encounter further issues, please provide additional details or screenshots to help us assist you!

**Comment by pderrenger** [comment]
Hi @joseant7231,

To use YOLOv5 for video recognition, install the required dependencies with:

```bash
pip install -r https://raw.githubusercontent.com/ultralytics/yolov5/master/requirements.txt
```

This installs PyTorch and all necessary libraries. For video processing specifically, ensure OpenCV is properly installed:

```bash
pip install opencv-python-headless
```

You can also clone the repository and install everything at once:

```bash
git clone https://github.com/ultralytics/yolov5
cd yolov5
pip install -r requirements.txt
```

For detailed setup instructions, check our [Quickstart guide](https://docs.ultralytics.com/quickstart/).
